src/uid.cpp

```diff
@@ -0,0 +1,4 @@
+#include "uid.h"
+
+template <>
+uid_factory<UID_ITEM> *uid_factory<UID_ITEM>::instance = nullptr;
```

Comment:
There's no reason for this to be a singleton, game can have an instance.  It's tied to game anyway.